src/guide/data-methods.md

```diff
@@ -0,0 +1,137 @@
+# Data Properties and Methods
+
+## Data Properties
+
+The `data` function for a component is called as part of creating a new component instance. It should return an object, which Vue wraps in its reactivity system and stores on the instance as `$data`. For convenience, any top-level properties of that object are also exposed directly via the component instance:
+
+```js
+const app = Vue.createApp({
+  data() {
+    return { count: 4 }
+  }
+})
+
+const vm = app.mount('#app')
+
+console.log(vm.$data.count) // => 4
+console.log(vm.count)       // => 4
+
+// Assigning a value to vm.count will also update $data.count
+vm.count = 5
+console.log(vm.$data.count) // => 5
+
+// ... and vice-versa
+vm.$data.count = 6
+console.log(vm.count) // => 6
+```
+
+These instance properties are only added when the instance is first created, so you need to ensure they are all present in the object returned by the `data` function. Where necessary, use `null`, `undefined` or some other placeholder value for properties where the desired value isn't yet available.
+
+It is possible to add a new property directly to the component instance without including it in `data`. However, because this property isn't backed by the reactive `$data` object it won't automatically be tracked by [Vue's reactivity system](reactivity.html).
+
+Vue uses a `$` prefix when exposing its own built-in APIs via the component instance, e.g. `$emit`. It also reserves the prefix `_` for internal properties. You should avoid using names for top-level `data` properties that start with either of these characters.
+
+## Methods
+
+You might be tempted to put functions in your `data` so that you can use them as methods:
```

Comment:
I've never encountered anyone who does this (putting functions in `data` as methods), no matter how much of a beginner they are so IMHO this might not be the best approach to introduce `methods`.

Reply:
Yeah, I made a similar observation here: https://github.com/vuejs/docs-next/pull/514/files/932839b6f612d79f299ab7e45604e0188d327e96#r491639058

I'll give it some more thought. I probably just need to delete most of that introduction and get to the point.

Reply:
While this is theoretically possible, I'm think this would introduce an anti-pattern and may be more harmful than helpful. However, I do think this would make an interesting blog post!

Reply:
I've rewritten the **Methods** section. A few small pieces remain but it's mostly new.

As a result, the sections on **Data Properties** and **Methods** aren't really linked anymore. They could be split into two pages, though they would be quite short compared to the rest of the guide.

I've given up on trying to explain that the properties created by `methods` aren't reactive. There are plenty of more important points to hit.